include/deal.II/lac/sparsity_pattern.h

```diff
@@ -149,6 +149,13 @@ namespace SparsityPatternIterators
      */
     size_type index () const;
 
+    /**
+     * Return global index in sparsity pattern.
```

Comment:
Say something like "This function returns the how-many'th entry within the entire sparsity pattern the current iterator points to. While the order of entries in a sparsity pattern is generally not important, this function allows indexing entries of the sparsity pattern and associated matrices using a linear index."